r/pkgdown/extra.js

```diff
@@ -59,6 +77,45 @@
       var empty_ul = $("#toc").find("ul").filter(":empty");
       empty_ul.remove();
     });
+
+    $(document).ready(function () {
+  
+      /**
+       * This replaces the package version number in the docs with a
+       * dropdown where you can select the version of the docs to view.
+       */
+
+        $pathStart = function(){
+    	  return window.location.origin + "/docs/";
+        }
+
+        $pathEnd  = function(){
+      	  var current_path = window.location.pathname;
+      	  return current_path.match("(?<=\/r).*");
+        }
+
+        // Load the versions JSON and construct the select items
+        $.getJSON("./versions.json", function( data ) {
```

Comment:
The "./versions.json", is that a relative path to the current location? Because that might not work if you have sub-directories in the docs (like for the articles) 

Or does that point to a "root" file? (in which case it would always be arrow.apache.org/versions.json once deployed, which might also not be correct?)

Reply:
Good catch; I guess it'd make the most sense to hard code it to point to the published version like this?

```suggestion
        $.getJSON("https://arrow.apache.org/docs/r/versions.json", function( data ) {
```

Reply:
Yeah, this part is annoying to test with an absolute url before it's actually deployed .. :) 

But, I think an absolute url is probably the easiest for now. I would maybe not put it in /docs/r/versions.json, as that is the stable docs location (it would get overwritten when doing a release and recreating the stable docs). Unless the file is not included in the actual arrow repo, and only in the arrow-site repo. Although I see now that I actually did the same for the sphinx site ... :)

Reply:
Just added another commit where I did it a little differently to construct a relative URL based on the fact that the only thing we know for sure is the structure that pkgdown has means that the final instance of `/r/`in the path will be where `versions.json` is hosted. I *think* it makes sense now!

Reply:
So does that mean that if you are on a page like https://arrow.apache.org/docs/5.0/r/articles/developing.html, that would point to https://arrow.apache.org/docs/5.0/r/versions.json? (taking the last /r, and appending versions.json)

Because then it still points to the versions.json of that version, and not the latest one (which can be updated)

Reply:
Argh, of course! Reverting to absolute URL...